src/main/java/seedu/cafectrl/parser/Parser.java

```diff
@@ -192,23 +194,27 @@ private static Command prepareEditPriceCommand(Menu menu, String arguments, Ui u
             String dishIndexText = matcher.group(dishIndexGroup).trim();
 
             // Check whether the index is empty
-            if (dishIndexText.equals("")) {
+            if (dishIndexText.isEmpty()) {
+                logger.warning("Empty dish index!");
                 return new IncorrectCommand(ErrorMessages.MISSING_DISH_IN_EDIT_PRICE, ui);
             }
 
             dishIndex = Integer.parseInt(dishIndexText);
 
             // Check whether the dish index is valid
             if (!menu.isValidDishIndex(dishIndex)) {
+                logger.warning("Invalid dish index!");
                 return new IncorrectCommand(ErrorMessages.INVALID_DISH_INDEX, ui);
             }
         } catch (NumberFormatException e) {
+            logger.warning("Invalid dish index type!");
```

Comment:
Perhaps you could pass the throwable for a more descriptive log message?
```suggestion
            logger.log(Level.WARNING, "Invalid dish index type!", e);
```